src/components/segmentedControl/index.tsx

```diff
@@ -144,35 +150,38 @@ const SegmentedControl = (props: SegmentedControlProps) => {
           isSelected={selectedSegment === index}
           activeColor={activeColor}
           inactiveColor={inactiveColor}
+          fullWidth={fullWidth}
           {...segments?.[index]}
         />
       );
     });
 
   return (
-    <View row center style={[styles.container, containerStyle, {borderRadius, backgroundColor}]}>
-      <Reanimated.View
-        style={[
-          styles.selectedSegment,
-          animatedStyle,
-          {borderColor: outlineColor, borderRadius, backgroundColor: activeBackgroundColor, borderWidth: outlineWidth}
-        ]}
-      />
-      {renderSegments()}
+    <View center={!fullWidth}>
+      <View row center style={[styles.container, containerStyle, {borderRadius, backgroundColor}]}>
+        <Reanimated.View
+          style={[
+            styles.selectedSegment,
+            animatedStyle,
+            {borderColor: outlineColor, borderRadius, backgroundColor: activeBackgroundColor, borderWidth: outlineWidth}
+          ]}
+        />
+        {renderSegments()}
+      </View>
     </View>
   );
 };
 
 const styles = StyleSheet.create({
   container: {
     backgroundColor: Colors.grey80,
-    height: Spacings.s7,
-    borderColor: Colors.grey60
+    borderColor: Colors.grey60,
+    paddingVertical: Spacings.s2,
+    borderWidth: BORDER_WIDTH
   },
   selectedSegment: {
-    height: Spacings.s7 - 2 * BORDER_WIDTH,
-    position: 'absolute',
-    backgroundColor: Colors.white
+    height: SEGMENT_HEIGHT,
```

Comment:
Seeing it again, I don't think that passing a fixed height is correct. 
It limits the component height and in case the content is bigger (bigger font for example) the component doesn't grow nicely.. the font will be cut

Reply:
Fixed :)